src/boot/primary.js

```diff
@@ -0,0 +1,220 @@
+//  primary.js
+//
+//  Created by Kalila L. on May 17th, 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+import { boot } from 'quasar/wrappers';
+
+import Log from '../modules/debugging/log.js';
+
+// more info on params: https://v2.quasar.dev/quasar-cli/boot-files
+export default boot(({ app, store, router, Vue }) => {
+    // MAIN APPLICATION INITIALIZATION
+
+    window.$ = window.jQuery = require('jquery');
+
+    function initializeAjax () {
+        Log.print('OTHER', 'INFO', 'Bootstrapping Ajax.');
+
+        window.$.ajaxSetup({
+            beforeSend: function (xhr) {
+                xhr.setRequestHeader('x-vircadia-error-handle', 'badrequest');
+                xhr.setRequestHeader('Authorization', 'Bearer ' + store.state.account.accessToken);
+            }
+        });
+    }
+
+    // END MAIN APPLICATION INITIALIZATION
+
+    // GLOBAL FUNCTIONS
+
+    function checkNeedsTokenRefresh () {
+        const currentTimestamp = Math.floor(new Date().getTime() / 1000); // in seconds
+        const sessionExpirationTime = parseInt(store.state.account.createdAt) + parseInt(store.state.account.expiresIn);
+
+        // console.info('createdAt', Object.prototype.toString.call(store.state.account.createdAt), 'expiresIn', Object.prototype.toString.call(store.state.account.expiresIn))
+        // console.info('currentTimestamp', currentTimestamp, Object.prototype.toString.call(currentTimestamp))
+        // console.info('sessionExpirationTime', sessionExpirationTime, Object.prototype.toString.call(sessionExpirationTime))
+
+        if (currentTimestamp > sessionExpirationTime ||
+            (currentTimestamp - store.state.globalConsts.SAFETY_BEFORE_SESSION_TIMEOUT) > sessionExpirationTime) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function attemptRefreshToken () {
+        window.$.ajax({
```

Comment:
suggest you return this value so the caller can potentially track the result (currently the caller will always get "undefined")

Reply:
Note that if you want to return the "true"/"false" that the sub-functions return then you may want to stick this in a Promise wrapper so it returns a standard promise rather than the jqXHR object